samza-core/src/test/java/org/apache/samza/serializers/model/serializers/TestIntermediateMessageSerde.java

```diff
@@ -126,4 +128,14 @@ public void testEndOfStreamMessageSerde() {
     assertEquals(de.getTaskName(), taskName);
     assertEquals(de.getVersion(), 1);
   }
+
+  @Test (expected = SamzaException.class)
+  public void testUserMessageSerdeException() {
+    IntermediateMessageSerde imserde = new IntermediateMessageSerde(new ObjectSerde());
+    IntermediateMessageSerde imserde2 = new IntermediateMessageSerde(new JsonSerdeV2<>());
```

Comment:
Instead of depending on `JsonSerdeV2` for this test, consider creating a mock `Serde` to do what you need.
In general, using mocks is helpful so that you don't rely on a specific implementation of something which is subject to change in the future. For example, what if `JsonSerdeV2` was changed to stop throwing an exception for certain kinds of data, or if it was changed to throw a different kind of exception than `SamzaException`?

Reply:
Good suggestion. I have updated to use a mocked serde.